src/mscorlib/src/System/Environment.cs

```diff
@@ -1052,8 +1086,27 @@ public static IDictionary GetEnvironmentVariables()
                 root[0]++;
             }
             return result;
+#else // PLATFORM_UNIX
+            // Use reflection to access the unix implementation in System.IO.FileSystem.DriveInfo.dll
+            Type driveInfoType = Type.GetType(
+                "System.IO.DriveInfo, System.IO.FileSystem.DriveInfo, Version=4.0.0.0, Culture=neutral, PublicKeyToken=" + AssemblyRef.MicrosoftPublicKeyToken,
```

Comment:
> we'd essentially need to recompile all of System.IO.FileSystem.DriveInfo.dll 

It does not need to be all of System.IO.FileSystem.DriveInfo.dll. It can be just a trivial wrapper around `Interop.Sys.GetAllMountPoints()`.